bin/_blackbox_common.sh

```diff
@@ -274,10 +284,37 @@ function enumerate_subdirs() {
     done
   done <"$listfile" | sort -u
 }
+ 
 
 # chdir to the base of the repo.
 function change_to_vcs_root() {
-  cd "$REPOBASE"
+  # if vcs_root not explicitly defined, use $REPOBASE
+  if [[ -z "$1" ]]; then
+    cd "$REPOBASE"
+
+  elif is_blackbox_repo "$1"; then
+    cd "$1"
+  
+  else
+    echo 'ERROR: $1 is not a blackbox Repo'
+    exit 1
+  fi
+
+}
+
+# $1 is a string pointing to a directory.  Outputs a
+# list of  valid blackbox repos,relative to $1
+function enumerate_blackbox_repos() {
+  if [[ -z "$1" ]]; then
+    echo "enumerate_blackbox_repos: ERROR: No Repo provided to Enumerate"
+  else
```

Comment:
Add an "exit 1" here.  The remaining code no longer needs to be in an else...fi indentation.